File: 1389.cpp
/*
	Author: Zoljargal Gantumur
	Runtime: 4 ms, faster than 63.86% of C++ online submissions
	Memory Usage: 8.6 MB, less than 26.49% of C++ online submissions
*/
class Solution {
public:
    vector<int> createTargetArray(vector<int>& nums, vector<int>& index) {
        vector<int> ans;
        ans.reserve(nums.size());
        for(int i=0; i<index.size(); i++){
            ans.insert(ans.begin() + index[i], nums[i]);
        }
        return ans;
    }
};

File: 1389.py
# Author: Zoljargal Gantumur
# Runtime: 28 ms, faster than 90.13% of Python3 online submissions
# Memory Usage: 14 MB, less than 20.18% of Python3 online submissions

class Solution:
    def createTargetArray(self, nums: List[int], index: List[int]) -> List[int]:
        ans = []
        for i, x in enumerate(index):
            ans.insert(x, nums[i])
        return ans
